### PR TITLE
feat: descruting application into multiple components

### PR DESCRIPTION
**Pull request by krivanek06**
Destructing app.components into multiple smaller components

### CODE REVIEW

**Comment by m-thompson-code** [comment]
There's a handful of things that will need to change for this PR, we'll sync outside of GitHub and consider a different approach

**Review by m-thompson-code — CHANGES_REQUESTED** [review]
Blocking for now

File: src/app/services/error-handler/error-interceptor.service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { catchError, Observable } from 'rxjs';
+import { RegistryError } from 'src/app/directives';
+
+@Injectable()
+export class ErrorInterceptorService implements HttpInterceptor {
+	constructor(private matSnackBar: MatSnackBar) {}
```
Comment (m-thompson-code) [inline src/app/services/error-handler/error-interceptor.service.ts:9]: Could we follow the pattern of using `private readonly` for our injectables? It should be common knowledge to not change the reference to an injected service, but I like the idea of marking which things should only be read and can't be overwritten

File: src/app/services/npm-registry/npm-registry.service.ts [hunk 2]
```diff
@@ -1,99 +1,96 @@
...
+import { inject, Injectable } from '@angular/core';
+import { format } from 'date-fns';
 import { catchError, forkJoin, map, Observable, of, tap } from 'rxjs';
 import { StorageService, StorageType } from '../storage';
-import { format } from 'date-fns';
 import { DownloadsPoint, DownloadsRange, RegistryData, Suggestion } from './npm-registry.model';
 
 @Injectable({
-  providedIn: 'root'
+	providedIn: 'root',
```
Comment (m-thompson-code) [inline src/app/services/npm-registry/npm-registry.service.ts:9]: github changes is showing this as 8 spaces for some reason. I assume this is github being weird or there actually is 8 spaces

**Review by m-thompson-code — CHANGES_REQUESTED** [review]
There's a lot of moving parts in this PR. I think it would be better to handle each individual task as its own PR to make approving code changes quicker. For instance, there doesn't seem to be a whole lot of code changes needed for date-range-picker component, but there's also stuff like adding prettier and an interceptor.

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "tabWidth": 2,
```
[the remark this answers is not in the document]
Comment (krivanek06) [reply .prettierrc:2]: The majority of files are already formatted with prettier, would it make sense to remove it now?

**Comment by krivanek06** [comment]
I don't see any added benefit of discarding this PR that is almost finished. I will fix the proposed changes

**Comment by m-thompson-code** [comment]
I'm going to close for now since we've made a lot of progress migrating each component in separate PRs

**Pull request closed without merging**